File: gma-next-gen-sdk-android-examples-main/java/nextgenexample/app/src/main/java/com/example/snippets/InterstitialAdSnippets.java
// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.example.snippets;

import android.app.Activity;
import android.util.Log;
import com.google.android.libraries.ads.mobile.sdk.common.AdRequest;
import com.google.android.libraries.ads.mobile.sdk.common.AdValue;
import com.google.android.libraries.ads.mobile.sdk.common.LoadAdError;
import com.google.android.libraries.ads.mobile.sdk.common.PreloadCallback;
import com.google.android.libraries.ads.mobile.sdk.common.PreloadConfiguration;
import com.google.android.libraries.ads.mobile.sdk.common.ResponseInfo;
import com.google.android.libraries.ads.mobile.sdk.interstitial.InterstitialAd;
import com.google.android.libraries.ads.mobile.sdk.interstitial.InterstitialAdEventCallback;
import com.google.android.libraries.ads.mobile.sdk.interstitial.InterstitialAdPreloader;

/** Java code snippets for the developer guide. */
public final class InterstitialAdSnippets {

  private static final String TAG = "InterstitialAdSnippets";

  // [START start_preload]
  private void startPreloading(String adUnitId) {
    AdRequest adRequest = new AdRequest.Builder(adUnitId).build();
    PreloadConfiguration preloadConfig = new PreloadConfiguration(adRequest);
    InterstitialAdPreloader.start(adUnitId, preloadConfig);
  }

  // [END start_preload]

  // [START set_buffer_size]
  private void setBufferSize(String adUnitId) {
    AdRequest adRequest = new AdRequest.Builder(adUnitId).build();
    PreloadConfiguration preloadConfig = new PreloadConfiguration(adRequest, 3);
    InterstitialAdPreloader.start(adUnitId, preloadConfig);
  }

  // [END set_buffer_size]

  // [START set_callback]
  private void startPreloadingWithCallback(String adUnitId) {
    PreloadCallback preloadCallback =
        // [Important] Don't call ad preloader start() or pollAd() within the PreloadCallback.
        new PreloadCallback() {
          @Override
          public void onAdFailedToPreload(String preloadId, LoadAdError adError) {
            Log.e(
                TAG,
                String.format(
                    "Interstitial preload ad %s failed to load with error: %s",
                    preloadId, adError.getMessage()));
            // [Optional] Get the error response info for additional details.
            // ResponseInfo responseInfo = adError.getResponseInfo();
          }

          @Override
          public void onAdsExhausted(String preloadId) {
            Log.i(TAG, "Interstitial preload ad " + preloadId + " is not available");
          }

          @Override
          public void onAdPreloaded(String preloadId, ResponseInfo responseInfo) {
            Log.i(TAG, "Interstitial preload ad " + preloadId + " is available");
          }
        };

    AdRequest adRequest = new AdRequest.Builder(adUnitId).build();
    PreloadConfiguration preloadConfig = new PreloadConfiguration(adRequest);
    InterstitialAdPreloader.start(adUnitId, preloadConfig, preloadCallback);
  }

  // [END set_callback]

  private void pollAd(String adUnitId) {
    // [START poll_ad]
    // Polling returns the next available ad and loads another ad in the background.
    final InterstitialAd ad = InterstitialAdPreloader.pollAd(adUnitId);
    // [END poll_ad]
  }

  // [START pollAndShowAd]
  private void pollAndShowAd(Activity activity, String adUnitId) {
    // Polling returns the next available ad and loads another ad in the background.
    final InterstitialAd ad = InterstitialAdPreloader.pollAd(adUnitId);

    // Interact with the ad object as needed.
    if (ad != null) {
      Log.d(TAG, "Interstitial ad response info: " + ad.getResponseInfo());
      ad.setAdEventCallback(
          new InterstitialAdEventCallback() {
            @Override
            public void onAdImpression() {
              Log.d(TAG, "Interstitial ad recorded an impression.");
            }

            @Override
            public void onAdPaid(AdValue value) {
              Log.d(
                  TAG,
                  "Interstitial ad onPaidEvent: "
                      + value.getValueMicros()
                      + " "
                      + value.getCurrencyCode());
            }
          });

      // Show the ad.
      ad.show(activity);
    }
  }

  // [END pollAndShowAd]

  // [START isAdAvailable]
  private boolean isAdAvailable(String adUnitId) {
    return InterstitialAdPreloader.isAdAvailable(adUnitId);
  }
  // [END isAdAvailable]
}

File: gma-next-gen-sdk-android-examples-main/kotlin/nextgenexample/app/src/main/java/com/example/snippets/InterstitialAdSnippets.kt
// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.example.snippets

import android.app.Activity
import android.util.Log
import com.google.android.libraries.ads.mobile.sdk.common.AdRequest
import com.google.android.libraries.ads.mobile.sdk.common.AdValue
import com.google.android.libraries.ads.mobile.sdk.common.LoadAdError
import com.google.android.libraries.ads.mobile.sdk.common.PreloadCallback
import com.google.android.libraries.ads.mobile.sdk.common.PreloadConfiguration
import com.google.android.libraries.ads.mobile.sdk.common.ResponseInfo
import com.google.android.libraries.ads.mobile.sdk.interstitial.InterstitialAdEventCallback
import com.google.android.libraries.ads.mobile.sdk.interstitial.InterstitialAdPreloader

/** Kotlin code snippets for the developer guide. */
private class InterstitialAdSnippets {

  // [START start_preload]
  private fun startPreloading(adUnitID: String) {
    val adRequest: AdRequest = AdRequest.Builder(adUnitID).build()
    val preloadConfig = PreloadConfiguration(adRequest)
    InterstitialAdPreloader.start(adUnitID, preloadConfig)
  }

  // [END start_preload]

  // [START set_buffer_size]
  private fun setBufferSize(adUnitID: String) {
    val adRequest: AdRequest = AdRequest.Builder(adUnitID).build()
    val preloadConfig = PreloadConfiguration(adRequest, bufferSize = 3)
    InterstitialAdPreloader.start(adUnitID, preloadConfig)
  }

  // [END set_buffer_size]

  // [START set_callback]
  private fun startPreloadingWithCallback(adUnitID: String) {
    val preloadCallback =
      // [Important] Don't call ad preloader start() or pollAd() within the PreloadCallback.
      object : PreloadCallback {
        override fun onAdFailedToPreload(preloadId: String, adError: LoadAdError) {
          Log.i(
            TAG,
            ("Interstitial preload ad $preloadId failed to load with error: ${adError.message}"),
          )
          // [Optional] Get the error response info for additional details.
          // val responseInfo = adError.responseInfo
        }

        override fun onAdsExhausted(preloadId: String) {
          Log.i(TAG, "Interstitial preload ad $preloadId is not available")
        }

        override fun onAdPreloaded(preloadId: String, responseInfo: ResponseInfo) {
          Log.i(TAG, "Interstitial preload ad $preloadId is available")
        }
      }
    val adRequest: AdRequest = AdRequest.Builder(adUnitID).build()
    val preloadConfig = PreloadConfiguration(adRequest)
    InterstitialAdPreloader.start(adUnitID, preloadConfig, preloadCallback)
  }

  // [END set_callback]

  private fun pollAd(adUnitID: String) {
    // [START poll_ad]
    // Polling returns the next available ad and loads another ad in the background.
    val ad = InterstitialAdPreloader.pollAd(adUnitID)
    // [END poll_ad]
  }

  // [START pollAndShowAd]
  private fun pollAndShowAd(activity: Activity, adUnitID: String) {
    // Polling returns the next available ad and loads another ad in the background.
    val ad = InterstitialAdPreloader.pollAd(adUnitID)

    // Interact with the ad object as needed.
    ad?.apply {
      Log.d(TAG, "Interstitial ad response info: ${this.getResponseInfo()}")
      this.adEventCallback =
        object : InterstitialAdEventCallback {
          override fun onAdImpression() {
            Log.d(TAG, "Interstitial ad recorded an impression.")
          }

          override fun onAdPaid(value: AdValue) {
            Log.d(TAG, "Interstitial ad onPaidEvent: ${value.valueMicros} ${value.currencyCode}")
          }
        }

      // Show the ad.
      ad.show(activity)
    }
  }

  // [END pollAndShowAd]

  // [START isAdAvailable]
  private fun isAdAvailable(adUnitID: String): Boolean {
    return InterstitialAdPreloader.isAdAvailable(adUnitID)
  }

  // [END isAdAvailable]

  private companion object {
    const val TAG = "InterstitialAdSnippets"
  }
}
